src/api/routes/address.ts

```diff
@@ -164,16 +164,25 @@ export function createAddressRouter(db: DataStore, chainId: ChainID): RouterWith
     if (!isValidPrincipal(stxAddress)) {
       return res.status(400).json({ error: `invalid STX address "${stxAddress}"` });
     }
-
-    const blockParams = getBlockParams(req, res, next);
-    const limit = parseTxQueryLimit(req.query.limit ?? 20);
-    const offset = parsePagingQueryInput(req.query.offset ?? 0);
-    const { results: txResults, total } = await db.getAddressTxs({
-      stxAddress: stxAddress,
-      limit,
-      offset,
-      ...blockParams,
-    });
+    let txResults: DbTx[] = [];
+    let total: number = 0,
+      limit = 0,
+      offset = 0;
+    try {
+      const blockParams = getBlockParams(req, res, next);
+      limit = parseTxQueryLimit(req.query.limit ?? 20);
+      offset = parsePagingQueryInput(req.query.offset ?? 0);
+      const addressTxsDb = await db.getAddressTxs({
+        stxAddress: stxAddress,
+        limit,
+        offset,
+        ...blockParams,
+      });
+      txResults = addressTxsDb.results;
+      total = addressTxsDb.total;
+    } catch (err: any) {
```

Comment:
Catching everything like this isn't good practice. Either throw and handle a specific error type, or refactor the functions to return something like an optional result rather than throw

Reply:
I sometimes use the safe await pattern to avoid of try/catch blocks and more deliberately catch errors https://github.com/DavidWells/safe-await